GG/src/ScrollPanel.cpp

```diff
@@ -0,0 +1,116 @@
+#include "../GG/ScrollPanel.h"
+
+#include <GG/DrawUtil.h>
+#include <GG/Flags.h>
+#include <GG/Scroll.h>
+#include <GG/Clr.h>
+#include <GG/ClrConstants.h>
+#include <GG/WndEvent.h>
+#include <GG/StyleFactory.h>
+
+#include <algorithm>
+
+namespace GG {
+
+    using std::max;
+    using std::min;
+
+    namespace {
+        // Minimum acceptable width for a scroll.
+        X MIN_SCROLL_WIDTH(14);
+
+        // Space to leave left of the vertical scroll.
+        X SCROLL_MARGIN_X(2);
+    }
+
+    void ScrollPanel::SizeMove(const Pt& ul, const Pt& lr)
+    {
+        // Store the new size for input to the layout.
+        Wnd::SizeMove(ul, lr);
+
+        // Lay out the children.
+        DoLayout();
+    }
+
+    void ScrollPanel::Render() {
+        FlatRectangle(UpperLeft(), LowerRight(), m_background_color, CLR_ZERO, 0);
+    }
+
+    ScrollPanel::ScrollPanel() : m_vscroll(0), m_content(0), m_background_color(CLR_ZERO) {}
+
+    ScrollPanel::ScrollPanel(X x, Y y, X w, Y h, Wnd* content)
+    : Wnd(x, y, w, h, INTERACTIVE), m_vscroll(0), m_content(content), m_background_color(CLR_ZERO)
```

Comment:
: to previous line, indent Wnd 4 spaces relative to previous line start, put each entry in the initializer list on its own line